lib/nerves_hub/socket.ex

```diff
@@ -11,26 +11,47 @@ defmodule NervesHub.Socket do
   def opts(nil), do: opts([])
 
   def opts(user_config) when is_list(user_config) do
-    ca_certs = Certificate.ca_certs()
-    cert = Nerves.Runtime.KV.get(@cert) |> Certificate.pem_to_der()
-    key = Nerves.Runtime.KV.get(@key) |> Certificate.pem_to_der()
+    ca_certs = user_config[:cacerts] || Certificate.ca_certs()
 
-    server_name =
+    {cert_key, cert_value} = cert(user_config)
+    {key_key, key_value} = key(user_config)
+
+    sni =
       (user_config[:server_name_indication] || @server_name)
-      |> to_charlist()
+    sni = if is_binary(sni), do: to_charlist(sni), else: sni
```

Comment:
i think `to_charlist/1` will just noop if the data is already a charlist:

```elixir
Interactive Elixir (1.7.3) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> to_charlist('abc')
'abc'
```

Reply:
This check is to prevent it from trying `to_charlist` on atoms